src/Dialogs/SmartPlaylistEditor.vala

```diff
@@ -324,7 +329,11 @@ public class Noise.SmartPlaylistEditorQuery : GLib.Object {
         rv.comparator = comparators.get (comparator_combobox.get_active ());
         if (needs_value ((SmartQuery.FieldType)field_combobox.get_active ())) {
             var value = Value (typeof (string));
-            value.set_string (value_entry.text);
+            if(rv.field == SmartQuery.FieldType.URI) {
+                value.set_string (value_entry.text.replace(" ", "%20"));
```

Comment:
Same here: Is there a reason why [GLib.Uri](https://valadoc.org/glib-2.0/GLib.Uri.html) cannot be used in this case?